packages/worker/src/runtime.ts

```diff
@@ -536,31 +536,51 @@ export class Runtime {
 
   protected checkHeapSizeLimit(): void {
     if (process.platform === 'linux') {
-      const cgroupMemoryConstraint = Number(
-        this.tryReadFileSync(/* cgroup v2 */ '/sys/fs/cgroup/memory.max') ??
-          this.tryReadFileSync(/* cgroup v1 */ '/sys/fs/cgroup/memory/memory.limit_in_bytes')
-      );
-
-      if (cgroupMemoryConstraint < os.totalmem() && cgroupMemoryConstraint < v8.getHeapStatistics().heap_size_limit) {
-        const totalMemInMb = toMB(os.totalmem(), 0);
-        const suggestedOldSpaceSizeInMb = toMB(os.totalmem() * 0.75, 0);
+      // References:
+      // - https://facebookmicrosites.github.io/cgroup2/docs/memory-controller.html
+      // - https://www.kernel.org/doc/Documentation/cgroup-v1/memory.txt
+      const cgroupMemoryConstraint =
+        this.tryReadNumberFileSync(/* cgroup v2 */ '/sys/fs/cgroup/memory.high') ??
+        this.tryReadNumberFileSync(/* cgroup v2 */ '/sys/fs/cgroup/memory.max') ??
+        this.tryReadNumberFileSync(/* cgroup v1 */ '/sys/fs/cgroup/memory/memory.limit_in_bytes');
+      const cgroupMemoryReservation =
+        this.tryReadNumberFileSync(/* cgroup v2 */ '/sys/fs/cgroup/memory.low') ??
+        this.tryReadNumberFileSync(/* cgroup v2 */ '/sys/fs/cgroup/memory.min') ??
+        this.tryReadNumberFileSync(/* cgroup v1 */ '/sys/fs/cgroup/memory/soft_limit_in_bytes');
+
+      const applicableMemoryConstraint = cgroupMemoryReservation ?? cgroupMemoryConstraint;
```

Comment:
nit: don't read the files for `cgroupMemoryReservation` if non of the memory constraint files were found.

Reply:
No. I really want to get both so that we can build the equivalent docker args below. I've seen a few cases recently where users had different value on request vs limit, where node process would get OOMkilled only once the host machine would be under pressure. Most often, the devs that were trying to diagnose these problem were not aware of how the pods had been configured, which slowed down research. I want to make it easier to find out these things.

Reply:
Initially, I was even considering printing out a warning whenever request and limit are different, but ended up not doing so since the risk is rather moot if heap is calculated based on _request_ (rather than limit).